app/src/main/java/com/yterletskyi/happyfriend/features/pin/ui/PinFragment.kt

```diff
@@ -21,6 +21,7 @@ class PinFragment : BaseBindingFragment<FragmentPinBinding>(
         super.onViewCreated(view, savedInstanceState)
 
         binding.pinKeyboardView.onButtonClicked = viewModel::input
+        viewModel.tempPin.value = args.pin
```

Comment:
Setting LiveData value from Fragments/Activities is usually bad idea. LiveDatas are only used to deliver value from ViewModels to views (Fragments, Activities etc). This is why we have public LiveData and not MutableLiveData in viewmodels (see `pinProgressLiveData` and `_pinProgressLiveData`).

Instead let's pass the pin code from args into viewmodel using `SavedStateHandle` (see example in `IdeasViewModel`)